vip-helpers/vip-caching.php

```diff
@@ -361,17 +361,17 @@ function wpcom_vip_old_slug_redirect() {
  * if it's called several times per request. This allows bypassing the db queries in favor of
  * the cache
  */
-function wpcom_vip_count_user_posts( $user_id ) {
+function wpcom_vip_count_user_posts( $user_id, $post_type = 'post', $public_only = false ) {
 	if ( ! is_numeric( $user_id ) ) {
 		return 0;
 	}
 
-	$cache_key   = 'vip_' . (int) $user_id;
+	$cache_key   = 'vip_' .  md5(wp_json_encode($post_type)) . '_' . (int) $user_id;
```

Comment:
Minor styling suggestion:
```suggestion
	$cache_key   = 'vip_' .  md5( wp_json_encode( $post_type ) ) . '_' . (int) $user_id;
```